### DISCUSSION TITLE
Conditional Jump depends on uninitialised value

### DISCUSSION THREAD

**Discussion by Shoesuke** [IIC2133-PUC/2024-1#166] [Tarea 2] [answered]
Holaa!, en la parte 1 paso bien todos los test cases pero me está dando el siguiente error en algunos test

![imagen](https://github.com/IIC2133-PUC/2024-1/assets/130422187/196d1fb0-10fa-42a9-810a-c56c0ce1be9b)

La verdad es que no tengo idea a que se debe ya que la linea 52 tiene la declaración del substring que uno lee desde el input y en la 75 solo uso strncpy por lo que apreciaría un poco de ayuda, aprovechando la ocasión quería saber si este tipo de errores descuenta en algo.

**Comment by CataAlamos** [comment, marked as answer]
Hola! Esto implica que tienes muchos errores de memoria que generan Conditional jump or move depends on uninitialised value.  Esto provoca que tu output de un valor correspondiente a NULL en varios eventos. Te recomiendo revisar con valgrind y solucionar estos problemas (Cuando aparece lo de conditional jump implica que el programa utilizara lo que se encuentre en memoria en ese momento, y por ende el resultado dependerá del sistema operativo, programas abiertos, que se ejecuto antes, etc).

**Reply by benjavicente** [reply]
Complementando la respuesta, Conditional Jump es cuando tienes memoria alocada, como se indica en la segunda parte del error, que no se le asigna ningún valor, y luego se utiliza ese "ningún valor" (uninitialised value) dentro de un if. Eso serí el comportamiento indefinido, tipo "¿qué pasa si mi condición es algo que realmente no está definido?". 

Como sale que el salto (if) está dentro de `strncopy`, Asegurate que todos los parámetros que le pasas a esa función estén ok, puedes printear esos valores por cada consulta. Es raro igual que el error no ocurra en test más chicos 🤔. Si subes tú código, podríamos echarle un ojo rápido.

---

El error descontaría en la categoría de errores de memoria.

**Reply by Shoesuke** [reply]
Holaa, se me fue lo más básico para debuggear que es hacer print de las variables, lo hice e identifiqué altiro el error, muchas gracias a ambos!!.
